### PR TITLE
Container Styles Implementation modified to suit the new changes in host-config 

### PR DESCRIPTION
**Pull request by Surezt22**
Container Styles Implementation modified to suit the new changes in host-config

### CODE REVIEW

File: source/community/reactnative/src/utils/host-config.js [hunk 1]
```diff
@@ -261,105 +269,48 @@ export class ActionsConfig {
...
-				this.foregroundColors.accent = this.getTextColorDefinitionOrDefault(obj.foregroundColors["accent"], { default: "#2E89FC", subtle: "#882E89FC" });
-				this.foregroundColors.good = this.getTextColorDefinitionOrDefault(obj.foregroundColors["good"], { default: "#54A254", subtle: "#DD54A254" });
-				this.foregroundColors.warning = this.getTextColorDefinitionOrDefault(obj.foregroundColors["warning"], { default: "#E69500", subtle: "#DDE69500" });
-				this.foregroundColors.attention = this.getTextColorDefinitionOrDefault(obj.foregroundColors["attention"], { default: "#CC3300", subtle: "#DDCC3300" });
-			}
-
-			this.highlightBackgroundColor = obj["highlightBackgroundColor"];
-			this.highlightForegroundColor = obj["highlightForegroundColor"];
-		}
+	parse(obj, type) {
```
Comment (pragadeeshk) [inline source/community/reactnative/src/utils/host-config.js]: Instead of null checking obj in every line, can we have single if else condition?

Comment (Surezt22) [reply source/community/reactnative/src/utils/host-config.js]: Made the changes

File: source/community/reactnative/src/utils/host-config.js [hunk 2]
```diff
@@ -261,105 +269,57 @@ export class ActionsConfig {
...
+		this.backgroundColor = defaultHostConfig["containerStyles"][type]["backgroundColor"];
+		this.foregroundColors.default = this.getTextColorDefinition(defaultHostConfig["containerStyles"][type].foregroundColors["default"]);
+		this.foregroundColors.accent = this.getTextColorDefinition(defaultHostConfig["containerStyles"][type].foregroundColors["accent"]);
+		this.foregroundColors.good = this.getTextColorDefinition(defaultHostConfig["containerStyles"][type].foregroundColors["good"]);
+		this.foregroundColors.warning = this.getTextColorDefinition(defaultHostConfig["containerStyles"][type].foregroundColors["warning"]);
+		this.foregroundColors.attention = this.getTextColorDefinition(defaultHostConfig["containerStyles"][type].foregroundColors["attention"]);
+		
 		if (obj) {
 			this.backgroundColor = obj["backgroundColor"];
```
Comment (BalajiR) [inline source/community/reactnative/src/utils/host-config.js]: Change this to `obj[type]["backgroundColor"]`

File: source/community/reactnative/src/utils/host-config.js [hunk 3]
```diff
@@ -261,105 +269,57 @@ export class ActionsConfig {
...
-				this.foregroundColors.light = this.getTextColorDefinitionOrDefault(obj.foregroundColors["light"], { default: "#FFFFFF", subtle: "#33000000" });
-				this.foregroundColors.accent = this.getTextColorDefinitionOrDefault(obj.foregroundColors["accent"], { default: "#2E89FC", subtle: "#882E89FC" });
-				this.foregroundColors.good = this.getTextColorDefinitionOrDefault(obj.foregroundColors["good"], { default: "#54A254", subtle: "#DD54A254" });
-				this.foregroundColors.warning = this.getTextColorDefinitionOrDefault(obj.foregroundColors["warning"], { default: "#E69500", subtle: "#DDE69500" });
-				this.foregroundColors.attention = this.getTextColorDefinitionOrDefault(obj.foregroundColors["attention"], { default: "#CC3300", subtle: "#DDCC3300" });
-			}
-
-			this.highlightBackgroundColor = obj["highlightBackgroundColor"];
-			this.highlightForegroundColor = obj["highlightForegroundColor"];
+			this.foregroundColors.default = this.getTextColorDefinition(obj[type].foregroundColors["default"]);
```
Comment (BalajiR) [inline source/community/reactnative/src/utils/host-config.js]: * Check the condition for the presence of `foregroundColors` prop
* Use `default` values from HostConfig for the missing foregroundColors.

**Pull request merged**